src/lib.rs

```diff
@@ -285,32 +292,194 @@ mod tests {
         ];
 
         #[rustfmt::skip]
-        let velocities = vec![
+        let values = vec![
             -9., -8., -7., -6., -5., -4.,
             -3., -2., -1.,  0.,  1.,  2.,
              3.,  4.,  5.,  6.,  7.,  8.,
         ];
 
-        // output buffers
-        let mut mean_vec = vec![0.0; 3];
-        let mut weight_vec = vec![0.0; 3];
+        (positions, values)
+    }
+
+    // apply_accum Tests
+    // -----------------
+
+    // based on numpy! (maybe look into a better approach)
+    // https://numpy.org/doc/stable/reference/generated/numpy.isclose.html
+    //
+    // maybe look into the approx crate?
+    fn _isclose(actual: f64, ref_val: f64, rtol: f64, atol: f64) -> bool {
+        let actual_nan = actual.is_nan();
+        let ref_nan = ref_val.is_nan();
+        if actual_nan || ref_nan {
+            actual_nan && ref_nan
+        } else {
+            (actual - ref_val).abs() <= (atol + rtol * ref_val.abs())
+        }
+    }
+
+    #[test]
+    fn apply_accum_errors() {
+        #[rustfmt::skip]
+        let positions = [
+             6.0,  7.0,
+            12.0, 13.0,
+            18.0, 19.0
+        ];
 
-        let points = PointProps::new(&positions, &velocities, None, 3_usize).unwrap();
+        #[rustfmt::skip]
+        let values = [
+            -9., -8.,
+            -3., -2.,
+             3.,  4.
+        ];
+
+        let bin_edges = [10.0, 5.0, 3.0];
+        let mut accum = Mean::new(bin_edges.len() - 1).unwrap();
+        let points = PointProps::new(&positions, &values, None, 3_usize).unwrap();
 
         let result = apply_accum(&mut accum, &points, None, &bin_edges);
-        assert_eq!(result, Ok(()));
-        // TODO check output
+        assert!(result.is_err());
+        assert!(result.unwrap_err().contains("monotonic"));
 
-        // should fail for non-monotonic bins
-        let unsorted_bin_edges = vec![25.0, 21.0, 17.0];
-        let result = apply_accum(&mut accum, &points, None, &unsorted_bin_edges);
+        let bin_edges = vec![3.0, 5.0, 0.0];
+        let result = apply_accum(&mut accum, &points, None, &bin_edges);
         assert!(result.is_err());
         assert!(result.unwrap_err().contains("monotonic"));
 
+        // should the following case be an error?
+        // let bin_edges = vec![0.0, 3.0, 3.0];
```

Comment:
Yes! But I think we should not worry about it until we implement a more sophisticated BinSpec trait and implementers.
```suggestion
        // TODO this should be an error
        // let bin_edges = vec![0.0, 3.0, 3.0];
```